bin/navsnmp

```diff
@@ -85,13 +86,25 @@ def snmp_printer(netbox):
         ipaddr = "ipv6:[%s]" % ipaddr
 
     print("# {}".format(netbox.sysname), file=sys.stderr)
-    print(
-        "-v{version} -c {community} {ipaddr}".format(
-            version=version,
-            community=profile.snmp_community,
-            ipaddr=ipaddr,
-        )
-    )
+
+    args = [f"-v{version}"]
+    if version != 3:
+        args.append(f"-c {profile.snmp_community}")
+    else:
+        conf = profile.configuration
+        args.extend(["-l", conf["sec_level"], "-u", conf["sec_name"]])
+        if conf.get("auth_protocol"):
+            args.extend(["-a", conf.get("auth_protocol")])
+        if conf.get("auth_password"):
+            args.extend(["-A", conf.get("auth_password")])
+        if conf.get("priv_protocol"):
+            args.extend(["-x", conf.get("priv_protocol")])
+        if conf.get("priv_password"):
+            args.extend(["-X", conf.get("priv_password")])
+
+    args.append(ipaddr)
```

Comment:
Uhm. These args does not exist in the argument parser. Is this code finished? What is the point of printing this stuff? Maybe a module docstring is in order. Confuse!

Reply:
Read the usage text on the help screen of the `navsnmp` program, @hmpf .

The purpose of this program is to produce valid arguments to NET-SNMP command line programs (`snmpcmd(1)`) from NAV SNMP management profiles.  It can be used to run manual SNMP queries from the command line of a NAV server like thus:

```sh
snmpwalk $(navsnmp some-box-from-nav.example.org) SNMPv2-MIB::system
```

Reply:
I would be tempted to move the description to a module docstring then setting argumenparser's description to __doc__.

Reply:
Sure, but the juicy usage examples are in the ArgumentParser `epilog`, and references `%(prog)s`, so that bit is too magical for the docstrings...